cmd/influxd/upgrade/config.go

```diff
@@ -168,36 +142,70 @@ func (cu *configUpgrader) save(config map[string]interface{}, path string) error
 
 // Credits: @rogpeppe (Roger Peppe)
 
-func (cu *configUpgrader) transform(x map[string]interface{}) map[string]interface{} {
+func (cu *configUpgrader) transform(v1Config map[string]interface{}) map[string]interface{} {
 	res := make(map[string]interface{})
 	for old, new := range cu.rules {
-		val, ok := cu.lookup(x, old)
-		if ok {
-			if transform, ok := cu.valueTransforms[new]; ok {
-				val = transform(val)
-			}
+		if val, ok := cu.lookup(v1Config, old); ok {
 			res[new] = val
 		}
 	}
 
+	// Special case: flip the value for pprof.
+	if val, ok := cu.lookup(v1Config, "http.pprof-enabled"); ok {
+		if b, ok := val.(bool); ok {
+			res["pprof-disabled"] = !b
+		}
+	}
+
+	// Special case: ensure query settings are valid.
+	fixQueryLimits(res)
+
 	return res
 }
 
-func (cu *configUpgrader) lookup(x map[string]interface{}, path string) (interface{}, bool) {
+// fixQueryLimits ensures that all query-related config settings are compatible
+// with the upgraded value of the 'query-concurrency' setting.
+func fixQueryLimits(v2Config map[string]interface{}) {
+	concurrencyVal, ok := v2Config["query-concurrency"]
+	if !ok {
+		return
+	}
+	var concurrency int64
+	switch c := concurrencyVal.(type) {
+	case int:
+		concurrency = int64(c)
+	case int32:
+		concurrency = int64(c)
+	case int64:
+		concurrency = c
```

Comment:
All these cases are paranoia on my part; I'm not sure when the config-loader would pick each type of int.